datastream-server-restli/src/test/java/com/linkedin/datastream/server/dms/TestDatastreamResources.java

```diff
@@ -153,6 +156,39 @@ public void testReadDatastream() throws Exception {
     Assert.assertEquals(ds, datastreamToCreate);
   }
 
+  @Test
+  public void testGettingInstanceHostingTask() {
+    DatastreamResources resource1 = new DatastreamResources(_datastreamKafkaCluster.getPrimaryDatastreamServer());
```

Comment:
If you decide to encapsulate the logic in the store, please add tests in TestZookeeperBackedDatastreamStore and you can use `zkclient` object to create the zookeeper state (good and bad cases) and test without mocking.